md2cf/api.py

```diff
@@ -212,6 +219,13 @@ def update_page(
 
         return self._put(f"content/{page.id}", json=update_structure)
 
+    def delete_page(self, confluence_page):
+        return self._delete(f"content/{confluence_page.id}")
+
+    def get_all_pages_from_parent_id(self, parent_id):
```

Comment:
Unfortunately this only returns the first level of descendants for the page, which means children of children are not deleted and are moved up to the parent. You'll have to implement a recursive descent where you build a tree of pages and delete from the leaves up.

Reply:
Good catch, didn't take that into account as it was not required for my use case. Changed this to use recursion